src/c++/perf_analyzer/genai-perf/tests/test_json_exporter.py

```diff
@@ -0,0 +1,282 @@
+
+from genai_perf.export_data.json_exporter import JsonExporter
+from genai_perf.llm_inputs.llm_inputs import OutputFormat, PromptSource
+
+
+class TestJsonExporter:
+    stats = {
+        "request_throughput": {"unit": "request per sec", "avg": "12.222026283281268"},
+        "request_latency": {
+            "unit": "ns",
+            "avg": "81694262.3143508",
+            "p99": "101977365.62",
+            "p95": "98139925.99999999",
+            "p90": "93780014.0",
+            "p75": "87927373.5",
+            "p50": "81988918.0",
+            "p25": "76813208.5",
+            "max": "180952000",
+            "min": "37899188",
+            "std": "11531917.69933574",
+        },
+        "time_to_first_token": {
+            "unit": "ns",
+            "avg": "20757771.22095672",
+            "p99": "29047316.52",
+            "p95": "27985451.7",
+            "p90": "25826318.599999998",
+            "p75": "22925623.5",
+            "p50": "20901523.0",
+            "p25": "18963453.5",
+            "max": "44816559",
+            "min": "10820287",
+            "std": "4220849.59851806",
+        },
+        "inter_token_latency": {
+            "unit": "ns",
+            "avg": "3860101.1100274306",
+            "p99": "5263723.220000001",
+            "p95": "5097144.25",
+            "p90": "5036146.1",
+            "p75": "4063794.25",
+            "p50": "3988754.5",
+            "p25": "3866785.25",
+            "max": "104056336",
+            "min": "619744",
+            "std": "1539307.1111368367",
+        },
+        "output_token_throughput": {
+            "unit": "tokens per sec",
+            "avg": "217.1010500160076",
+        },
+        "output_token_throughput_per_request": {
+            "unit": "tokens per sec",
+            "avg": "221.26925312949",
+            "p99": "320.769756629708",
+            "p95": "297.8740207961169",
+            "p90": "277.716572304638",
+            "p75": "237.54220501729532",
+            "p50": "214.42163973683745",
+            "p25": "198.40160210092222",
+            "max": "362.2711821112862",
+            "min": "79.15736875418017",
+            "std": "36.9078750549695",
+        },
+        "num_output_token": {
+            "unit": "tokens",
+            "avg": "17.763097949886106",
+            "p99": "22.0",
+            "p95": "20.0",
+            "p90": "20.0",
+            "p75": "19.0",
+            "p50": "18.0",
+            "p25": "17.0",
+            "max": "25",
+            "min": "3",
+            "std": "1.7190889855652978",
+        },
+        "num_input_token": {
+            "unit": "tokens",
+            "avg": "550.0523917995445",
+            "p99": "552.24",
+            "p95": "550.0",
+            "p90": "550.0",
+            "p75": "550.0",
+            "p50": "550.0",
+            "p25": "550.0",
+            "max": "553",
+            "min": "550",
+            "std": "0.34348803287081675",
+        },
+    }
+    args = {
```

Comment:
I would monkeypatch `sys.argv` using `pytest.mark.parameterize` instead of manually specifying all values. 
For instance,
```python
@pytest.mark.parameterize(
    "extra_args",
    [
        ...
    ],
)
def test_generate_json(self, extra_args):
    args = ["genai-perf", "--model", "test_model"] + extra_args
    monkeypatch.setattr("sys.argv", args)
    args, _ = parser.parse_args()
```

This allows us to test `JsonExporter` with actual namespace object our parser generates.

Reply:
This is great. Moved to using the parser.